src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -2221,7 +2221,8 @@ public void visitTypeCast(JCTypeCast tree) {
         // the conversion.
         if (!tree.clazz.type.isPrimitive() &&
            !types.isSameType(tree.expr.type, tree.clazz.type) &&
-           types.asSuper(tree.expr.type, tree.clazz.type.tsym) == null) {
+           types.asSuper(tree.expr.type, tree.clazz.type.tsym) == null &&
+           tree.clazz.getTag() != TYPEINTERSECTION) {
```

Comment:
I think that if the right form of the AST is one that have only two type casts, then that's what the fix should be doing

Reply:
I revised the patch to adjust the AST. Now it has only two type casts instead of three.